src/main/java/seedu/address/model/BookShelf.java

```diff
@@ -20,9 +20,11 @@
 public class BookShelf implements ReadOnlyBookShelf {
 
     private final UniqueBookList books;
+    private String key;
```

Comment:
why do you need the key here?

Reply:
there should be no changes to this file.

Reply:
I should get key from the bookshelf's data send it to modelmanager

Reply:
The key shouldn't be in the bookshelf data.

Reply:
I think it is in the ReadOnlyBookShelf